src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -500,6 +500,89 @@ public void testDeleteBucketWithObjects() throws Exception {
         }
     }
 
+    @Test
+    public void testDeleteBucketWithBackgroundTasks() throws Exception {
+        Assume.assumeTrue("ECS version must be at least 3.8", ecsVersion != null && ecsVersion.compareTo("3.8") >= 0);
+
+        String bucketName = getTestBucket() + "-delete";
+        String bucketNameNotBeingDeleted = getTestBucket() + "-notBeingDeleted";
+
+        client.createBucket(bucketName);
+        client.putObject(bucketName, "foo", "bar", null);
+        client.createBucket(bucketNameNotBeingDeleted);
+        client.putObject(bucketNameNotBeingDeleted, "foo", "bar", null);
+
+        // check the status of a bucket without any deletion tasks.
+        BucketDeletionStatus statusNotBeingDeleted = client.getBucketDeletionStatus(bucketNameNotBeingDeleted);
+        Assert.assertEquals("EmptyBucketStatus should not be found", "NOT_FOUND", statusNotBeingDeleted.getStatus());
+        client.deleteBucket(new DeleteBucketRequest(bucketNameNotBeingDeleted, true));
+
+        // delete a bucket which is empty or not, the first status should always be PENDING.
+        client.deleteBucket(new DeleteBucketRequest(bucketName, true));
+        BucketDeletionStatus status = client.getBucketDeletionStatus(bucketName);
+        // check the status and dates of the bucket delete task, other entries are not given.
+        Assert.assertEquals("The initial status of a bucket which is being deleting should be PENDING.", status.getStatus(), "PENDING");
+        Assert.assertEquals("created date should be the same as lastUpdated date in EmptyBucketStatus.", status.getCreated(), status.getLastUpdated());
+
```

Comment:
would be good to somehow test the remaining status fields (counts, size, etc.)

Reply:
still wait for dev to confirm if objectCount/totalSize are null.